### PR TITLE
Fix papercuts issue #7099 - silo spew on shutdown

### PR DESCRIPTION
**Pull request by MarkNickeson**
- Focused on silo shutdown code path and reducing verbosity of LogLevel.Information
- Changed many LogLevels from Information to Debug
- Added IsEnabled conditional around modified code blocks to mitigate string build overhead
- Cleaned up a small number of logging related ctors, fields and logging context/type strings

[Shutdown log before](https://gist.github.com/MarkNickeson/33b1472a923c193fb935ab6a2f462c67)
[Shutdown log after](https://gist.github.com/MarkNickeson/24e9fd759503be7a19fc858d84b732b6)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/7733)

### CODE REVIEW

**Comment by ReubenBond** [comment]
cc @benjaminpetit

**Comment by ReubenBond** [comment]
Looks like there are some test failures: https://orleans.visualstudio.com/GitHub%20-%20PR%20Builds/_build/results?buildId=52366&view=ms.vss-test-web.build-test-results-tab&runId=1056910&resultId=100098&paneView=debug

**Comment by benjaminpetit** [comment]
Thanks @MarkNickeson !

Overall LGTM, but I think it would be great if we could log if we are shutting down gracefully or not. I think in 3.x we do, but we lost that in main.

File: src/Orleans.Runtime/Silo/Silo.cs [hunk 1]
```diff
@@ -454,11 +457,18 @@ public async Task StopAsync(CancellationToken cancellationToken)
...
-                    logger.Info(ErrorCode.WaitingForSiloStop, "Waiting {0} for termination to complete", pause);
+                    logger.Debug(ErrorCode.SiloStopInProgress, "Silo shutdown in progress. Waiting for shutdown to be completed.");
+                }
+                var pause = TimeSpan.FromSeconds(1);                
+
+                while (!this.SystemStatus.Equals(SystemStatus.Terminated))
+                {                    
+                    if (logger.IsEnabled(LogLevel.Debug))
+                    {
+                        logger.Debug(ErrorCode.WaitingForSiloStop, "Silo shutdown still in progress...", pause);
```
Comment (ReubenBond) [inline src/Orleans.Runtime/Silo/Silo.cs:480]: The pause parameter isn't being logged here, but it's being passed to the log call.

**Pull request merged**